client/my-sites/plans-grid/hooks/use-get-free-trial-slug-for-plan.ts

```diff
@@ -0,0 +1,22 @@
+import { PLAN_HOSTING_TRIAL_MONTHLY, PlanSlug, isBusinessPlan } from '@automattic/calypso-products';
+import { isNewHostedSiteCreationFlow } from '@automattic/onboarding';
+import { useCallback } from 'react';
+import { useSelector } from 'calypso/state';
+import { isUserEligibleForFreeHostingTrial } from 'calypso/state/selectors/is-user-eligible-for-free-hosting-trial';
```

Comment:
Cannot have Calypso state references in plans-grid folder. Think of this as a package already.

Reply:
It's outside the `npm-ready` folder though, and I see we already have other hooks in that folder with Calypso references.

Reply:
Yeah, we don't want to introduce new ones. Sorry if existence of folder directed towards thinking that it's ok to do so outside. Folder only there for maintenance/organisational reasons

Reply:
Oh, alright. I'll change it 👍